app/src/androidTest/java/org/mozilla/fenix/ui/robots/NavigationToolbarRobot.kt

```diff
@@ -0,0 +1,57 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+@file:Suppress("TooManyFunctions")
+
+package org.mozilla.fenix.ui.robots
+
+import android.net.Uri
+import androidx.test.espresso.Espresso.onView
+import androidx.test.espresso.action.ViewActions.click
+import androidx.test.espresso.action.ViewActions.pressImeActionButton
+import androidx.test.espresso.action.ViewActions.replaceText
+import androidx.test.espresso.matcher.ViewMatchers
+import androidx.test.platform.app.InstrumentationRegistry
+import androidx.test.uiautomator.By
+import androidx.test.uiautomator.UiDevice
+import androidx.test.uiautomator.Until
+import org.mozilla.fenix.R
+import org.mozilla.fenix.helpers.TestAssetHelper.waitingTime
+import org.mozilla.fenix.helpers.click
+
+/**
+ * Implementation of Robot Pattern for the URL toolbar.
+ */
+class NavigationToolbarRobot {
+    class Transition {
+
+        val mDevice = UiDevice.getInstance(InstrumentationRegistry.getInstrumentation())
+
+        fun enterURLAndEnterToBrowser(url: Uri, interact: BrowserRobot.() -> Unit): BrowserRobot.Transition {
+            mDevice.waitForIdle()
```

Comment:
we've discovered these generic device wait statements don't really do much. We need to remove them altogether (if they don't help) or, in the event the test has time flakiness at this point, replace them with an mDevice. wait on a specific element like on line 41:  ```mDevice.wait(Until.findObject(By.text("Menu")), waitingTime)```